da/celestia/celestia.go

```diff
@@ -189,30 +199,29 @@ func (c *DataAvailabilityLayerClient) SubmitBatch(batch *types.Batch) da.ResultS
 			},
 		}
 	}
-
-	blockBlob, err := blob.NewBlobV0(c.config.NamespaceID.Bytes(), data)
-	if err != nil {
+	if len(data) > DefaultMaxBytes {
 		return da.ResultSubmitBatch{
 			BaseResult: da.BaseResult{
 				Code:    da.StatusError,
-				Message: err.Error(),
+				Message: "Batch size bigger than maximum blob size",
 			},
 		}
 	}
-	blobs := []*blob.Blob{blockBlob}
-
-	estimatedGas := DefaultEstimateGas(uint32(len(data)))
-	gasWanted := uint64(float64(estimatedGas) * c.config.GasAdjustment)
-	fees := c.calculateFees(gasWanted)
-	c.logger.Debug("Submitting to da blob with size", "size", len(blockBlob.Data), "estimatedGas", estimatedGas, "gasAdjusted", gasWanted, "fees", fees)
 
 	for {
 		select {
 		case <-c.ctx.Done():
 			c.logger.Debug("Context cancelled")
 			return da.ResultSubmitBatch{}
 		default:
-			txResponse, err := c.rpc.SubmitPayForBlob(c.ctx, math.NewInt(fees), gasWanted, blobs)
+			gasPrice := float64(-1)
+			if c.config.GasPrices >= 0 {
```

Comment:
I think I got it from rollkit example to indicate is not set, but I actually removed gasPrices as input parameter for the submit() function, because it can be get later on from the config anyway, and no need to pass as a parameter.